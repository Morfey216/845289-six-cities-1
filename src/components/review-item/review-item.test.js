import React from 'react';
import renderer from 'react-test-renderer';
import ReviewItem from './review-item';

const mockReviewData = {
  id: 1,
  user: {
    id: 16,
    name: `Mollie`,
    avatarUrl: `https://es31-server.appspot.com/six-cities/static/avatar/7.jpg`,
    isPro: true
  },
  rating: 4,
  review: `Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius`,
  date: `2019-06-23T05:59:03.102Z`
};

it(`ReviewItem correctly renders`, () => {
  const tree = renderer
      .create(<ReviewItem reviewData={mockReviewData}/>)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
