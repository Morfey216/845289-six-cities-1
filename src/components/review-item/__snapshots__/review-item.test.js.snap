// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem correctly renders 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="https://es31-server.appspot.com/six-cities/static/avatar/7.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Mollie
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "40%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius
    </p>
    <time
      className="reviews__time"
      dateTime="2019-06-23T05:59:03.102Z"
    >
      June 2019
    </time>
  </div>
</li>
`;
